Don't set the LIBRARY_PATH environment variable.

b11z opened the pull request:
LIBRARY_PATH is for gcc linker scripts and seems unnecessary here.
It's causing issues because LIBRARY_PATH is set to a value including a bare
colon (e.g. ":/blah/blah") which causes GCC to search the current directory.
That breaks the compilation of some python modules (specifically guppy)

nolman commented:
Are there quotes in the LIBRARY_PATH? Colon is valid in LIBRARY_PATH?

From GCC manual:
LIBRARY_PATH
The value of LIBRARY_PATH is a colon-separated list of directories, much like PATH.

Have you tried this fork with ruby's rgeo (and ensure the geo lib gets correctly linked to rgeo)?

What is the literal value of LIBRARY_PATH when you build?

b11z commented:
We end up with a LIBRARY_PATH that's:
LIBRARY_PATH=:/app/.heroku/vendor/lib:/app/vendor/boost/stage/lib:/app/vendor/vowpalwabbit/.libs

It's the colon at the front that's the problem. GCC does a LIBRARY_PATH.split(":"), and that list contains the empty string, which gets treated as "." and causes GCC to pickup files from the current working directory.

I could alter the script so it doesn't tack the colon on front (tho i don't really see where it's happening), so it'd be
LIBRARY_PATH=/app/.heroku/vendor/lib:/app/vendor/boost/stage/lib:/app/vendor/vowpalwabbit/.libs
That'd be the most surgical fix. But I see no mention of LIBRARY_PATH in gdal/geos/proj docs and never came across it in a GCC build environment before -- was guessing it was cruft.

FWIW it seems to have built OK without it -- I tested the ruby and python environments. 
$ heroku run -a staging-...-bmb rails c
irb(main):004:0> RGeo::Geos.supported?
=> true
irb(main):005:0> RGeo::Geos.version
=> "3.4.1"

nolman commented:
Problematic code:
https://github.com/nolman/heroku-geo-buildpack/blob/master/bin/compile#L55

If you force rgeo recompile (could just point it to master) and then check the same thing on addresses and get the same result (RGeo::Geos.supported? #=> true) that is also fine.

That said we also mess up LD_LIBRARY_PATH with the leading colon, so maybe worth fixing still.

Not opposed to removing LIBRARY_PATH if unused, just want to avoid future debugging issues.

Commit:
Dont put colon on front of CPATH, LIBRARY_PATH, LD_LIBRARY_PATH

Make set-default-env act gracefully if appending to an empty
environment variable. Having a colon on the front of LIBRARY_PATH
is problematic as it causes GCC to search the cwd.
